### ISSUE TITLE
When using a project and partial builds, cross file errors not emitted

### ISSUE THREAD

**Issue by nmalaguti** [ivogabe/gulp-typescript#104]
If there are 2 files
- a.ts
- b.ts

**a.ts**
```typescript
export function foo(first: string, second: number) {
    console.log(first);
    console.log(second);
}
```

**b.ts**
```typescript
import {foo} from "./a.ts"

foo("hello", 5);
```

My gulpfile uses something like [gulp-cached](https://github.com/wearefractal/gulp-cached) to only pass through changed files when watching the directory. If I change the method signature in a.ts to `foo(first: number, second: number)`, it compiles a.ts correctly, but doesn't emit an error for b.ts.

This is because the gulp-typescript project caches the output for b.ts and just reemits it instead of re-compiling it.

This can be fixed by using the full language service to pass in changed files and emit all files/errors that need to be recompiled as a result. This should actually significantly speed up incremental compilation since most of the files will already be loaded by the typescript compiler.

I'd be happy to help out transitioning from using createProgram to createLanguageService if that is something you would support.

**Comment by ivogabe** [comment]
You can't use gulp-cached with gulp-typescript. TypeScript needs to have all files to do type checking. With gulp-cached there is no difference between a deleted file and a file that didn't change. In you example, it looks like `b.ts` was removed. Only `a.ts` is now compiled and there are no errors in that file. `b.ts` is **not** emitted.

The cache is only used when no file has been changed, and in the future, it will also be used for single file transpilation.

I'm already using incremental compilation, so using the full language service won't change it. Unchanged files (when you're not using gulp-cached) aren't re-parsed, but are type-checked and compiled. That speeds up the compilation.
